docs/introduction/zns.md

```diff
@@ -140,26 +138,26 @@ the current number of open and active zones.
 
 ### Zone Append
 
-The NVMe specifications allow a device controller to execute commands present
+The NVMe specification allows a device controller to execute commands present
 in the several submission queues available in any order. This has implications
-for the host IO stack, namely, even if the host submits write commands directed
-at a zone sequentially, the commands may be reordered before they are processed
-and violate the sequential write requirement, resulting in errors. Host software
-can avoid such error by limiting the number of write commands outstanding per
-zone to one. This can potentially result in poor performance, especially for
-workloads issuing mostly small write operations.
+for the host IO stack, namely, if the host submits multiple write commands to
+a zone sequentially, the commands may be reordered by the controller before
+processing. This violates the sequential write requirement resulting in an error.
```

Comment:
This is a tricky one that has a lot of nasty details... E.g. a PCI device with a single queue will guarantee that commands will be executed in order since commands are always taken from the submission queue in order. ZNS does not define any constraint on the controller to maintain command ordering *between* submission queues. That is the issue here. So the wording is a little strong... But I am not sure how to properly soften it without a lot of explanations...

Reply:
Can't a single queue case also result in reordering if let's say a cmd gets blocked internally due to lack of a firmware resource? Anyway, I've left this as-is for now.